src/main/java/teammates/ui/controller/InstructorCourseAddAction.java

```diff
@@ -91,7 +91,7 @@ private void createCourse(CourseAttributes course) {
 
             InstructorAttributes curInstructor = logic.getInstructorForGoogleId(course.id, account.googleId);
 
-            if (Logic.isCourseArchived(course.id, curInstructor.googleId)) {
+            if (curInstructor.isArchived) {
```

Comment:
Perhaps that can be done on the JSTL side instead, someone could introduce getters for the instructors during a jsp to jstl migration.